File: SWEA/D3/1208. ［S／W 문제해결 기본］ 1일차 － Flatten/［S／W 문제해결 기본］ 1일차 － Flatten.java
import java.util.Scanner;
import java.io.FileInputStream;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
class Solution
{
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=10;

		for(int test_case = 1; test_case <= T; test_case++)
		{
			int N = sc.nextInt();
			int[] arr = new int[100];
			for (int i = 0; i < 100; i++) {
				arr[i] = sc.nextInt();
			}

			int cnt = 0;
			int max_index;
			int min_index;
			int max;
			int min;
			int answer = 0;
			for (int i = 0; i < 1000; i++) {
				max = Arrays.stream(arr).max().getAsInt();
				min = Arrays.stream(arr).min().getAsInt();
				List<Integer> list = Arrays.stream(arr).boxed().collect(Collectors.toList());
				max_index = list.indexOf(max);
				min_index = list.indexOf(min);
				arr[max_index] = arr[max_index] - 1;
				arr[min_index] = arr[min_index] + 1;
				cnt++;

				if (cnt == N) {
					break;
				}
				if (Math.abs(arr[max_index] - arr[min_index]) == 0) {
					break;
				}
			}
			// result 를 돌면서
			max = Arrays.stream(arr).max().getAsInt();
			min = Arrays.stream(arr).min().getAsInt();
			List<Integer> list = Arrays.stream(arr).boxed().collect(Collectors.toList());
			max_index = list.indexOf(max);
			min_index = list.indexOf(min);
			answer = Math.abs(arr[max_index] - arr[min_index]);
			
			System.out.println("#" + test_case + " " + answer);
		}
	}
}

File: SWEA/D3/1208. ［S／W 문제해결 기본］ 1일차 － Flatten/［S／W 문제해결 기본］ 1일차 － Flatten.py
T = 10
for test_case in range(1, T + 1):
    N = int(input())
    land = list(map(int, input().split()))
    cnt = 0
    # 만약 cnt 횟수가 N이랑 같아지면 그땐 끝내야 한다.
    # 다 수행하고 난 뒤의 최댓값과 최솟값의 차이.
    # 아니면 할 때마다 min 값이랑 max 값을 계속 구해서 max는 -1하고
    # min 은 +1 하고?
    for i in range(1000):
        max_index = land.index(max(land))
        min_index = land.index(min(land))
        land[max_index] -= 1
        land[min_index] += 1

        cnt += 1
        if cnt == N:
            break
        if abs(max(land)-min(land))  == 0:
            break

    answer = 0
    answer = abs(max(land)- min(land))
    print(f'#{test_case} {answer}')